client/lib/sites-list/list.js

```diff
@@ -196,18 +193,11 @@ SitesList.prototype.update = function( sites ) {
 
 	this.markCollisions( sites );
 	this.data = sites.map( function( site ) {
-		var siteObj, result;
-
-		if ( sitesMap[ site.ID ] ) {
-			// Update existing Site object
-			siteObj = sitesMap[ site.ID ];
-			result = siteObj.set( site );
```

Comment:
Could we maybe update `Site.prototype.set` to explicitly check for the icon case instead? I feel like this may cause regressions